src/data/finset/basic.lean

```diff
@@ -1886,6 +1886,18 @@ mem_map.trans $ by simp only [exists_prop]; refl
   b ∈ s.map f.to_embedding ↔ f.symm b ∈ s :=
 by { rw mem_map, exact ⟨by { rintro ⟨a, H, rfl⟩, simpa }, λ h, ⟨_, h, by simp⟩⟩ }
 
+lemma map_perm {σ : equiv.perm α} (hs : {a | σ a ≠ a} ⊆ s) : s.map (σ : α ↪ α) = s :=
```

Comment:
```suggestion
/-- If the only elements outside `s` are those left fixed by `σ`, then mapping by `σ` has no effect -/
lemma map_perm {σ : equiv.perm α} (hs : {a | σ a ≠ a} ⊆ s) : s.map (σ : α ↪ α) = s :=
```
Is it worth also showing the stronger:
```lean
lemma map_perm_eq_of_subset {σ : equiv.perm α} (hs : s.map (σ : α ↪ α) ⊆ s) : s.map (σ : α ↪ α) = s :=
```

Reply:
Don't know, but it's done

Reply:
This doesn't seem to be the most natural condition for the lemma named `map_perm`. Do we already have the set `σ.fixed`? The this would be `map_perm_of_compl_fixed_subset` and we could also have `map_perm_of_subset_fixed`.

Reply:
`σ.support` is almost what we want, but its a finset. Perhaps we should rename it to `σ.fin_support` and introduce `support` as `{a | σ a ≠ a}`?

Reply:
The thing is that we already have `{a | σ a ≠ a}` but only in a fintype, so the point of this lemma was to use `equiv.perm.support` without writing it as `σ.support`.

Reply:
Ok, fair enough. If we want to fix `(fin_)support`, let's do that in a later PR.

bors merge